pkgs/tools/security/cryptomator/default.nix

```diff
@@ -98,7 +98,13 @@ in stdenv.mkDerivation rec {
     cp ${src}/dist/linux/common/application-vnd.cryptomator.vault.xml $out/share/mime/packages/application-vnd.cryptomator.vault.xml
   '';
 
-  nativeBuildInputs = [ autoPatchelfHook maven makeWrapper wrapGAppsHook jdk ];
+  nativeBuildInputs = [
+    autoPatchelfHook
+    maven
+    makeWrapper
+    (wrapGAppsHook.override { makeBinaryWrapper = makeWrapper; })
```

Comment:
Why is the override needed for cryptomator?

Reply:
https://github.com/NixOS/nixpkgs/pull/172320#issuecomment-1122288164

We could also remove the single quotes, but then we would be implicitly relying on the binary wrapper being in scope, which isn't great.

Reply:
So makeBinaryWrapper end up in cryptomator's nativeBuildInputs without? That should not happen.

Reply:
> So makeBinaryWrapper ends up in cryptomator's nativeBuildInputs without? That should not happen.

Yes, because it's a propagated build input of the wrapGApps hook (that is how setup hooks work). I agree that's not great, but I don't know how we should fix it. This is a limitation of Bash more than anything.

> The derivation doesn't use gappsWrapperArgs.

It doesn't need to. It first creates the `cryptomator` executable as a wrapper over java, and then wrapGAppsHook wraps that wrapper automatically in the fixup phase.

Reply:
> Yes, because it's a propagated build input of the wrapGAppsHook (that is how setup hooks work). I agree that's not great, but I don't know how we should fix it. This is a limitation of Bash more than anything.

Just use an absolute path to makeBinaryWrapper in the hook and don't propagate it.

Reply:
makeBinaryWrapper is also a setup hook, i.e. a Bash script that defines functions. Functions aren't scoped in Bash, so we would need to do something like

```bash
wrapGApp() (
    local program="$1"
    shift 1
    source @makeWrapper@
    wrapProgram "$program" "${gappsWrapperArgs[@]}" "$@"
)
```

in order not to expose the functions to the global scope. Maybe it could work.

Reply:
But makeWrapper also depends on dieHook, so we'd also need to source that explicitly... it's kind of awkward.

Reply:
We could call one of the bash functions makeBinaryWrapper and use a boolean argument for wrapGAppsHook

Reply:
Then let's call the other makeShellWrapper so people can decide which one they want to use if they feel like it. That would make this override superfluous.